File: backend/app/models/place.py
from __future__ import annotations

from pydantic import BaseModel, Field

from app.utils.enums import PlaceCategory  # noqa: TC001


class Coordinates(BaseModel):
    """A data model to represent a geographic coordinates."""

    lat: float = Field(..., description="Latitude of the coordinates.")

    lng: float = Field(..., description="Longitude of the coordinates.")


class Place(BaseModel):
    """A data model to represent a single point of interest in the nomad loop."""

    name: str = Field(..., description="Name of the place.")

    address: str = Field(..., description="Address of the place.")

    category: PlaceCategory = Field(
        ...,
        description="Category of the place, e.g., living, working, etc.",
    )

    url: str | None = Field(
        None,
        description="URL of the place, e.g., website or social media link.",
    )
    price: float | None = Field(
        None,
        description="Price of the place, e.g., cost of living or provided service.",
    )

    coordinates: Coordinates = Field(
        ...,
        description="Geographic coordinates of the place.",
    )

File: frontend/src/types/interfaces/place.ts
import { Coordinates } from '@src/types/interfaces/coordinates';
import { PlaceCategory } from '@src/utils/enums';

export interface Place {
  /** Name of the place */
  name: string;

  /** Address of the place */
  address: string;

  /** Category of the place, e.g., living, working, etc. */
  category: PlaceCategory;

  /** URL of the place, e.g., website or social media link */
  url?: string | null;

  /** Price of the place, e.g., cost of living or provided service */
  price?: number | null;

  /** Geographic coordinates of the place */
  coordinates: Coordinates;
}
